component-lib/declarations/README.md

```diff
@@ -0,0 +1,32 @@
+# What is the "declarations" folder for?
+
+This "declarations" folder exists to fix broken TypeScript type definitions for the Apache Arrow project.
+
+The apache-arrow 0.17.0 npm package fails to compile in modern TypeScript. The issue is tracked here: https://issues.apache.org/jira/browse/ARROW-8394
+
+As a workaround, we provide a patched set of Arrow type declarations in this folder.
```

Comment:
Just checking. Does it make sense to create a separate package for described temporary types rather than including them into our package? Just might be cleaner as a temp measure.

Reply:
These don't actually become part of the package that's uploaded to npm - they're just a formality to get TypeScript to compile the project. (They aren't code, they're just type definitions.) I'd prefer as few packages as possible, so I'd argue against this unless we were using it in multiple places.